### PR TITLE
Move Resources implementation to their package 🎗

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

GitResource, ImageResource, … are PipelineResource implementation and
are not really part of the API. At least, they shouldn't be part of
the `pkg/apis/pipeline` package.

This moves those to their own packages under
`pkg/apis/resource/v1alpha1`:
- `cloudevent` for `CloudEventResource`
- `cluster` for `ClusterResource`
- `git` for `GitResource`
- `image` for `ImageResource`
- `pullrequest` for `PullRequestResource`
- `storage` for all the storage resources `GCSResource`,
  `BuildGCSResource` and `ArtifactBucket`, `ArtifactPVC`

This renames the struct (to make lint and I happy 😌) and moves
`v1alpha1.ResourceFromType` to `resource.FromType`

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/cc @afrittoli @sbwsg @bobcatfish @skaegi 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
v1alpha1 PipelineResource implementation package is moved out of the Pipeline API package
```

### CODE REVIEW

**Comment by vdemeester** [comment]
Opened #2105 to reduce the size of this PR :angel:

File: pkg/apis/resource/v1alpha1/storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+		if _, ok := mountedSecrets[volName]; !ok {
+			storageVol = append(storageVol, gcsSecretVolume)
+			mountedSecrets[volName] = ""
+		}
+	}
+	return storageVol
+}
+
+// of the step will include name.
```
Comment (sbwsg) [inline pkg/apis/resource/v1alpha1/storage/storage.go]: Possibly missing comment lines here?

Comment (vdemeester) [reply pkg/apis/resource/v1alpha1/storage/storage.go]: oh yeah indeed.. what happened there…

**Comment by bobcatfish** [comment]
Awesome! Thanks for cleaning this up @vdemeester :D

Quick question: I don't quite understand why there are changes in `third_party` in this PR as well?

**Comment by vdemeester** [comment]
> Awesome! Thanks for cleaning this up @vdemeester :D
> 
> Quick question: I don't quite understand why there are changes in `third_party` in this PR as well?

I don't either… it got changed when I ran `./hack/update-codegen.sh` :sweat:

File: pkg/apis/pipeline/v1alpha1/storage_resource.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2019 The Tekton Authors
+Copyright 2019-2020 The Tekton Authors
```
Comment (dibyom) [inline pkg/apis/pipeline/v1alpha1/storage_resource.go]: should this just be 2020?

**Pull request merged**